xbmc/addons/AddonRepos.cpp

```diff
@@ -0,0 +1,347 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "AddonRepos.h"
+
+#include "Addon.h"
+#include "AddonDatabase.h"
+#include "AddonManager.h"
+#include "AddonSystemSettings.h"
+#include "CompileInfo.h"
+#include "Repository.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+#include <vector>
+
+using namespace ADDON;
+
+/**********************************************************
+ * CAddonRepos
+ *
+ */
+
+namespace
+{
+
+std::vector<RepoInfo> LoadOfficialRepoInfos()
+{
+  const std::vector<std::string> officialAddonRepos =
+      StringUtils::Split(CCompileInfo::GetOfficialAddonRepos(), ',');
+
+  std::vector<RepoInfo> officialRepoInfos;
+  RepoInfo newRepoInfo;
+
+  for (const auto& addonRepo : officialAddonRepos)
+  {
+    const std::vector<std::string> tmpRepoInfo = StringUtils::Split(addonRepo, '|');
+    newRepoInfo.m_repoId = tmpRepoInfo.front();
+    newRepoInfo.m_origin = tmpRepoInfo.back();
+    officialRepoInfos.emplace_back(newRepoInfo);
+  }
+
+  return officialRepoInfos;
+}
+
+} // unnamed namespace
+
+static std::vector<RepoInfo> officialRepoInfos = LoadOfficialRepoInfos();
```

Comment:
here i am not sure what to do - just place the static function into the unnamed namespace? 😀

Reply:
During my tests (about binary addons) does also not a empty namespace help :unamused:, stupid thing there on gcc/clang since them implemented export as default.

What it can:
```cpp
namespace ADDON
{
static std::vector<RepoInfo> officialRepoInfos = LoadOfficialRepoInfos();
}
```
as the `using namespace ADDON;` above in *.cpp not does it

or 

```cpp
// Inside class on *.h
static std::vector<RepoInfo> officialRepoInfos

// on *.cpp
std::vector<RepoInfo> CAddonRepos::officialRepoInfos
```

Reply:
decided to use `namespace ADDON {}`